Unexpected requirement for 'sudo make' on Fedora (possibly also affects CentOS/RHEL)

This may be relevant to #577   

As of 2018-04-06 (samtools v1.8) you will get the errors
shown below if you do not use 'sudo' to run 'make':

============================================================
bam_tview_curses.c:48:2: warning: #warning "No curses
library is available; tview with curses is disabled."
[-Wcpp]
. . .
misc/maq2sam.c: In function ‘main’:
misc/maq2sam.c:215:41: error: ‘SAMTOOLS_VERSION’ undeclared
(first use in this function)
   fprintf(stderr, "Version: %s\n", SAMTOOLS_VERSION);
misc/maq2sam.c:215:41: note: each undeclared identifier is
    reported only once for each function it appears in
make: *** [misc/maq2sam-long.o] Error 1
misc/maq2sam.c: In function ‘main’:
misc/maq2sam.c:215:41: error: ‘SAMTOOLS_VERSION’ undeclared
    (first use in this function)
fprintf(stderr, "Version: %s\n", SAMTOOLS_VERSION);
misc/maq2sam.c:215:41: note: each undeclared identifier is
    reported only once for each function it appears in
make: *** [misc/maq2sam-long.o] Error 1
============================================================

The ‘SAMTOOLS_VERSION’ errors (when running make as a 
non-root user) are new; I have not experienced these prior 
to v1.8

I install samtools (using prefix) as a non-root user in 
a non-root-owned subdir under /opt 

This work-around shown below (run in the samtools source directory) 
eliminates these errors on Fedora 23, and leaves everything in the source  
directory owned by the (non-root) installer.  Admittedly this work-around is
not ideal as it requires sudo privileges.

Subsequent 'make test' and/or 'make install' (to a user-accessible directory 
specified by prefix when configure was run) commands can be run as a 
non-root user.

############################################################

sudo make

grpNameDefault="$( id -g -n "${USER}" )"

sudo chown -R "${USER}:${grpNameDefault}" "$(pwd)"

############################################################

--- a/misc/maq2sam.c
+++ b/misc/maq2sam.c
@@ -1,6 +1,6 @@
 /*  maq2sam.c -- convert MAQ output to SAM.
 
-    Copyright (C) 2008, 2009 Genome Research Ltd.
+    Copyright (C) 2008, 2009, 2018 Genome Research Ltd.
 
     Author: Heng Li <lh3@sanger.ac.uk>
 
@@ -30,7 +30,7 @@ DEALINGS IN THE SOFTWARE.  */
 #include <inttypes.h>
 #include <stdlib.h>
 #include <assert.h>
-#include "version.h"
+#include "../version.h"
 
 //#define MAQ_LONGREADS
 

--- a/misc/wgsim.c
+++ b/misc/wgsim.c
@@ -1,6 +1,6 @@
 /* The MIT License
 
-   Copyright (c) 2008 Genome Research Ltd (GRL).
+   Copyright (c) 2008, 2018 Genome Research Ltd (GRL).
                  2011 Heng Li <lh3@live.co.uk>
 
    Permission is hereby granted, free of charge, to any person obtaining
@@ -39,7 +39,7 @@
 #include <ctype.h>
 #include <string.h>
 #include <zlib.h>
-#include "version.h"
+#include "../version.h"
 #include "htslib/kseq.h"
 #include "htslib/hts_os.h"
 KSEQ_INIT(gzFile, gzread)

